lib/js_of_ocaml/dom.ml

```diff
@@ -391,6 +398,24 @@ let preventDefault ev =
   then (Js.Unsafe.coerce ev)##preventDefault
   else (Js.Unsafe.coerce ev)##.returnValue := Js.bool false
 
+let createCustomEvent ?bubbles ?cancelable ?detail typ =
+  let opt_iter f = function
+    | None -> ()
+    | Some x -> f x
+  in
+  let opts = Unsafe.obj [||] in
+  opt_iter (fun x -> opts##.bubbles := bool x) bubbles;
+  opt_iter (fun x -> opts##.cancelable := bool x) cancelable;
+  opt_iter (fun x -> opts##.detail := some x) detail;
+  let constr :
+      (   ('a, 'b) #customEvent Js.t Event.typ
+       -> < detail : 'b opt prop > t
+       -> ('a, 'b) customEvent t)
+      constr =
+    Unsafe.global##._CustomEvent
```

Comment:
Where does the `_` in `_CustomEvent` come from?

Reply:
From `js_of_ocaml` documentation:

> Since ocaml doesn't allows method name to start with capitalised letter, we can add an _

Since `CustomEvent` constructor starts with the capital letter, we should prefix it with an underscore.

Reply:
Another example from the docs:

> Using a JS constructor, ex: F:
> Write in .ml:

`let f = Js.Unsafe.global##._F`